site/migration/01-migration.md

```diff
@@ -0,0 +1,257 @@
+The functionality is offered by `kpt pkg update` in a more user friendly fashion.
+Running the new `kpt pkg update` command on a package which has subpackages, simply
+means traversing the package hierarchy, and for each package encountered, update
+the package if needed. If you want to declaratively update nested subpackages,
+you can declare the desired version in the upstream section of the respective
+subpackage and invoking `kpt pkg update` on the root package will take care of updating
+all the nested subpackages.
+
+### Changes to Functions
+
+#### `kpt fn run` deprecated
+
+The functionality of kpt fn run command in `v0.39.0` is split into two different CLI
+commands that execute functions corresponding to two fundamentally different approaches:
+
+`kpt fn render:` Executes the pipeline of functions declared in the package and its
+subpackages. This is a declarative way to run functions.
+
+`kpt fn eval:` Executes a given function on the package. The image to run and the
+functionConfig is specified as CLI argument. This is an imperative way to run
+functions.
+
+#### Function Config
+
+As a result of these changes, we no longer need function configs to hold
+`config.kubernetes.io/function` annotation. Functions can be declared in the
+pipeline section of Kptfile and invoked using `kpt fn render`. The declared function
+can point to the function config using `configPath` argument.
+
+#### Function Results
+
+In both `render` and `eval`, structured results can be enabled using the `--results-dir` flag.
+Please refer to the [function results] section for more information on the new structure of function results.
+
+### Changes to Setters
+
+`Setters` and `substitutions` no longer follow the complex and verbose OpenAPI format.
+The setters and substitutions are simplified to use new and simple setter pattern
+comments. Creating a setter is as simple as adding a line comment to the desired field.
+Users need not add any `OpenAPI` definitions. Please refer to [apply-setters] for
+the information about new syntax for setters and how to apply setter values, this is
+equivalent to the `kpt cfg set` in `v0.39.0`.
+[Setter Inheritance] works as usual in kpt `v1.0`.
+
+Here is the [simple example] of setter syntax transformation.
+
+#### Setter validations deprecated
+
+We want to keep the authoring experience of setters as simple as possible and
+hence [OpenAPI validations] and [required setters] feature offered by `v0.39.0` setters
+is no longer offered in `v1.0` version of kpt. However, we are working on providing
+an easy way to achieve similar functionality retaining the simplicity of setters,
+which is scoped for post `v1.0` release. Stay tuned.
+
+#### Auto-setters deprecated
+
+[Auto-setters] feature is deprecated in `v1.0` version of kpt. Since the setters are
+migrated to a new and simple declarative version, package consumers can easily
+declare all the setter values and render them all at once.
+
+### Changes to Live
```

Comment:
In each section refer to relevant sections of kpt.dev (reference or book) for more information.

Reply:
Updated.